lib/active_merchant/billing/gateways/elavon.rb

```diff
@@ -352,30 +358,45 @@ def add_line_items(xml, level_3_data)
         }
       end
 
-      def add_stored_credential(xml, options)
+      def add_stored_credential(xml, payment_method, options)
+        return unless options[:stored_credential]
         network_transaction_id = options.dig(:stored_credential, :network_transaction_id)
         case
-        when network_transaction_id.nil?
-          return
         when network_transaction_id.to_s.include?('|')
           oar_data, ps2000_data = options[:stored_credential][:network_transaction_id].split('|')
-          xml.ssl_oar_data oar_data unless oar_data.nil? || oar_data.empty?
-          xml.ssl_ps2000_data ps2000_data unless ps2000_data.nil? || ps2000_data.empty?
+          xml.ssl_oar_data oar_data unless oar_data.nil? || oar_data.empty? || payment_method.is_a?(String)
```

Comment:
Could this be updated to `oar_data.blank?` instead of having `oar_data.nil? || oar_data.empty?` ?